src/bidiMapper/domains/context/browsingContextImpl.ts

```diff
@@ -440,32 +452,34 @@ export class BrowsingContextImpl {
       return;
     }
 
-    if (!this.#targetDefers.documentInitialized.isFinished) {
-      this.#targetDefers.documentInitialized.reject('Document changed');
+    if (this.#targetDefers.documentInitialized.isFinished) {
+      this.#targetDefers.documentInitialized = new Deferred<void>();
+    } else {
+      this.#logger?.(LogType.browsingContexts, 'Document changed');
     }
-    this.#targetDefers.documentInitialized = new Deferred<void>();
 
-    if (!this.#targetDefers.Page.navigatedWithinDocument.isFinished) {
-      this.#targetDefers.Page.navigatedWithinDocument.reject(
+    if (this.#targetDefers.Page.navigatedWithinDocument.isFinished) {
+      this.#targetDefers.Page.navigatedWithinDocument = new Deferred<Protocol.Page.NavigatedWithinDocumentEvent>();
+    } else {
+      this.#logger?.(
+        LogType.browsingContexts,
         'Document changed'
       );
     }
```

Comment:
After #462 is merged, this piece should be removed completely.

Reply:
Thanks for the heads up.